### PR TITLE
Many small improvements to cmake and travis configs

### PR DESCRIPTION
**Pull request by gfr10598**
Add test and build badges (in readme)
Add clang config for travis (without coverage)
Fix install-protobuf so that it always uses gcc.
Pass coverage flags from travis into cmake.

---
This change is [Reviewable](https://reviewable.io/reviews/m-lab/tcpinfo_lib/9)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,10 +4,26 @@ cmake_minimum_required (VERSION 2.8.6)
...
+if (NOT DEFINED CXX_COMPILER)
+  set(CXX_COMPILER "clang++")
+  set(C_COMPILER "clang")
+endif (NOT DEFINED CXX_COMPILER)
+
+if (COVERAGE_FLAGS STREQUAL "")
+# If we aren't instrumenting for coverage, then add optimization flags.
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
+else (COVERAGE_FLAGS STREQUAL "")
```
Comment (pboothe) [inline CMakeLists.txt:18]: Cmake's if/else/endif syntax is really weird. Nothing for you to change, just noting it.

File: .travis.yml [hunk 2]
```diff
@@ -46,19 +70,12 @@ cache:
...
 before_script:
 # Check if protobuf compiler is present, and load if it is not.
+# Note that this builds with gcc, regardless of which compiler is specified.
 - sudo ./install-protobuf.sh
 
 script:
 # Build and run tests.
 - cd build
-- cmake -DCMAKE_CXX_COMPILER=$CXX_COMPILER -DCMAKE_C_COMPILER=$C_COMPILER .. && cmake --build .
+- cmake -DCMAKE_CXX_COMPILER=$CXX_COMPILER -DCMAKE_C_COMPILER=$C_COMPILER -DCOVERAGE_FLAGS="$COVERAGE_FLAGS" .. && cmake --build .
```
Comment (pboothe) [inline .travis.yml]: Is the && necessary? Doesn't this die when any command fails to return true?

File: install-protobuf.sh [hunk 3]
```diff
@@ -13,8 +13,13 @@ if [ $? -ne 0 ] ; then
   else
     echo "Install protobuf from web."
     git clone https://github.com/google/protobuf.git
+    # For unknown reason, compiling protobuf on travis-ci breaks with clang.
+    # So we override to always use gcc.
+    CC=gcc
+    CXX=g++
     cd protobuf && git checkout v3.1.0
-    ./autogen.sh && ./configure --prefix=/usr && make && sudo make install && sudo ldconfig
+    ./autogen.sh && ./configure --prefix=/usr && make && \
```
Comment (pboothe) [inline install-protobuf.sh]: This line is getting pretty silly. This section could also be written as:

``` bash
(
  # For unknown reason, compiling protobuf on travis-ci breaks with clang.
  # So we override to always use gcc.
  CC=gcc
  CXX=g++
  set -e
  cd protobuf
  git checkout v3.1.0
  /autogen.sh
  ./configure --prefix=/usr
  make
  sudo make install
  sudo ldconfig
)
```

Whether to make that change is a matter of taste which I leave up to you.

File: install-protobuf.sh [hunk 4]
```diff
@@ -13,8 +13,13 @@ if [ $? -ne 0 ] ; then
   else
     echo "Install protobuf from web."
     git clone https://github.com/google/protobuf.git
+    # For unknown reason, compiling protobuf on travis-ci breaks with clang.
```
Comment (pboothe) [inline install-protobuf.sh]: reasons

**Review by pboothe — APPROVED** [review]
LGTM, optional style and less-optional spelling comments only.

File: CMakeLists.txt (hunk 1 above)
Comment (gfr10598) [reply CMakeLists.txt:18]: Ack.  FYI, I think the bodies are optional for else and endif, but I'll check another time.

File: .travis.yml (hunk 2 above)
Comment (gfr10598) [reply .travis.yml]: Should do.  I'll try it.

File: install-protobuf.sh (hunk 4 above)
Comment (gfr10598) [reply install-protobuf.sh]: Done.

File: install-protobuf.sh (hunk 3 above)
Comment (gfr10598) [reply install-protobuf.sh]: Done.

**Pull request merged**